src/jit/morph.cpp

```diff
@@ -15704,7 +15714,7 @@ void Compiler::fgMorphStmts(BasicBlock* block, bool* lnot, bool* loadw)
                          (call->IsFastTailCall() && (compCurBB->bbJumpKind == BBJ_RETURN) &&
                           (compCurBB->bbFlags & BBF_HAS_JMP)));
         }
-        else if (block != compCurBB)
+        else if (block != compCurBB && ((tree->gtOper == GT_CALL) && tree->AsCall()->IsTailCall()))
```

Comment:
The `insertedBB` is in `Compiler::fgMorphCall`. This code is in `Compiler::fgMorphStmts()` so I would have to make some kind of member flag. I don't think that is worth it.

To be honest, I merely applied the asserts to the predicate because it didn't appear to be related to the GC poll logic in my scenario.